src/graph/executor/algo/SubgraphExecutor.cpp

```diff
@@ -65,6 +66,14 @@ folly::Future<Status> SubgraphExecutor::getNeighbors() {
         }
         vids_.clear();
         return handleResponse(std::move(resp));
+      })
+      .thenError(folly::tag_t<std::bad_alloc>{},
```

Comment:
How about other Executor?

Reply:
> How about other Executor?

I am trying add more check, including storaged's execution path.

Reply:
Could we catch all bad allocation exception in one place like `Scheduler`?

Reply:
> Could we catch all bad allocation exception in one place like `Scheduler`?

I tried, but experimented show that exception occured in folly's Executor can be catched in thenError, but outside seems not able to catch it.

Reply:
It's no conflict. As I known, `Scheduler` will call `execute` function of each Executor and got the `Future` of execution. So, we could process bad allocation exception from this `Future`.
WDYT?

Reply:
> It's no conflict. As I known, `Scheduler` will call `execute` function of each Executor and got the `Future` of execution. So, we could process bad allocation exception from this `Future`. WDYT?

Yes, no conflict, I have also added catch in Scheduler, see Scheduler.cpp